### PR TITLE
DEVENGAGE-472 - Adding file config and logging to the dotnet SDK

### PR DESCRIPTION
**Pull request by ronanwatkins**

### CODE REVIEW

File: resources/sdk/pureclouddotnet/templates/ApiClient.mustache [hunk 1]
```diff
@@ -248,6 +249,9 @@ namespace {{packageName}}.Client
                 }
             }
 
+            if ((int)response.StatusCode < 200 || (int)response.StatusCode >= 300)
```
Comment (carnellj) [inline resources/sdk/pureclouddotnet/templates/ApiClient.mustache:255]: Why do you have to do the cast here.  Doesnt .NET do autoboxing of values.  If not I would be surprised if their Integer object does not have a toInt() method. on it.

File: resources/sdk/pureclouddotnet/templates/Configuration.mustache [hunk 2]
```diff
@@ -62,6 +66,47 @@ namespace {{packageName}}.Client
...
+            if (AutoReloadConfig)
+            {
+                ThreadStart childref = new ThreadStart(RunConfigChecker);
+                Console.WriteLine("In Main: Creating the Child thread");
+                Thread childThread = new Thread(childref);
+                childThread.Start();
+            }
+        }
+
+        private void applyConfigFromFile()
```
Comment (carnellj) [inline resources/sdk/pureclouddotnet/templates/Configuration.mustache:120]: Why is this function empty?

File: resources/sdk/pureclouddotnet/templates/Configuration.mustache [hunk 3]
```diff
@@ -62,6 +66,47 @@ namespace {{packageName}}.Client
...
+        {
+
+        }
+
+        public static void RunConfigChecker()
+        {
+            while (true)
+            {
+                Console.WriteLine("runConfigChecker");
+                Thread.Sleep(2000);
```
Comment (carnellj) [inline resources/sdk/pureclouddotnet/templates/Configuration.mustache]: Just to double check...... I am assuming this thread. Sleep was running off the thread you spun off for checking the config.  Also, don't hard code the value using a magic number.  Put 2000 in a constant at the top of the file so we know exactly why this parameter is there.

File: resources/sdk/pureclouddotnet/templates/Logger.mustache [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+                    }
+                }
+            }
+        }
+
+        private static string requestBodyToString(object requestBody)
+        {
+            if (requestBody != null)
+            {
+            //     if (requestBody.GetType() == typeof(String))
```
Comment (carnellj) [inline resources/sdk/pureclouddotnet/templates/Logger.mustache]: Why is this code commented out?  Remove uncommented code unless you are planning to use it.

File: resources/sdk/pureclouddotnet/templates/Logger.mustache [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+        public int StatusCode { get; set; }
+
+        public string RequestBody { get; set; }
+
+        public string ResponseBody { get; set; }
+
+        public string AsString(LogFormat logFormat, bool logRequestBody, bool logResponseBody)
+        {
+            RequestHeaders["Authorization"] = "[REDACTED]";
```
Comment (carnellj) [inline resources/sdk/pureclouddotnet/templates/Logger.mustache]: Nice

**Review by carnellj — APPROVED** [review]
Couple of small changes.  Please fix or let me know why they need to stay

File: resources/sdk/pureclouddotnet/templates/Configuration.mustache (hunk 2 above)
Comment (ronanwatkins) [reply resources/sdk/pureclouddotnet/templates/Configuration.mustache:120]: I think you're viewing the version from the first commit. If you look at the changes from https://github.com/purecloudlabs/platform-client-sdk-common/pull/153/files it should be all up-to-date

**Review by WitLi07 — APPROVED** [review]

**Review by jacob-shaw-genesys — APPROVED** [review]

**Pull request merged**